server/src/main/java/org/opensearch/gateway/remote/model/RemoteIndexMetadata.java

```diff
@@ -41,6 +44,24 @@ public class RemoteIndexMetadata extends AbstractClusterMetadataWriteableBlobEnt
     public static final String INDEX = "index";
 
     private IndexMetadata indexMetadata;
+    private RemoteStoreEnums.PathType pathType;
+    private RemoteStoreEnums.PathHashAlgorithm pathHashAlgo;
+    private String fixedPrefix;
+
+    public RemoteIndexMetadata(
+        final IndexMetadata indexMetadata,
+        final String clusterUUID,
+        final Compressor compressor,
+        final NamedXContentRegistry namedXContentRegistry,
+        final RemoteStoreEnums.PathType pathType,
+        final RemoteStoreEnums.PathHashAlgorithm pathHashAlgo,
+        final String fixedPrefix
```

Comment:
is there a possibility to keep in check the number of ctors we are creating for this class?

Reply:
I can delete one constructor as that is used in tests only